core/src/stores/postgres.rs

```diff
@@ -1859,25 +1894,21 @@ impl Store for PostgresStore {
             token_count: None,
         };
 
-        // TODO(KW_SEARCH_INFRA): make title/mime_type not optional.
-        // Upsert the data source node if title and mime_type are present. Otherwise, we skip the
-        // upsert.
-        if should_upsert_node {
-            self.upsert_data_source_node(
-                UpsertNode {
-                    node_id: &document.document_id,
-                    node_type: &NodeType::Document,
-                    timestamp: document.timestamp,
-                    title: &document.title,
-                    mime_type: &document.mime_type,
-                    parents: &document.parents,
-                },
-                data_source_row_id,
-                document_row_id,
-                &tx,
-            )
-            .await?;
-        }
+        self.upsert_data_source_node(
```

Comment:
do we have guarantee we were always in the case `should_upsert_node = true` since migration?

Reply:
We were skipping it before to avoid having default values on things upserted by connectors. Now connectors send the correct values, and the backfill has been run. Since all select will start  relying on data_sources_nodes we must ensure nodes are always inserted , even if mimetype/title are not passed. We may still have this ( so cases where we had should_upsert_node = false) as it's not mandatory in public api, but it will now insert with default mimetpye and title.

Reply:
ack